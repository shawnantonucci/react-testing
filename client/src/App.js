import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  state = {
    isOn: false,
  };
  render() {
    return (
      <div className="App">
        <h1 className="display">Switch</h1>
      </div>
    );
  }
}

export default App;
