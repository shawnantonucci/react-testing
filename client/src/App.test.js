import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {
  
  
  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('matches snapshot', () => {
    const snapshot = renderer.create(<App />).toJSON();

    expect(snapshot).toMatchSnapshot();
  })

  it('should render a h1 with a display class', () => {
    const wrapper = shallow(<App />);
    
    expect(wrapper.find('h1.display').length).toBe(1);
  })

  it('should be off by default', () => {
    const wrapper = shallow(<App />);
    const instance = wrapper.instance();

    expect(instance.state.isOn).toBe(false);
  })
})

/*

- does it render without blowing up
- does it render the right UI
- state/props handled correctly
- events/methods working correctly

Specifications

- should be "off" by default

*/