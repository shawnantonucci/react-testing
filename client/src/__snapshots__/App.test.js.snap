// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="display"
  >
    Switch
  </h1>
</div>
`;
